lib/handlebars/compiler/compiler.js

```diff
@@ -323,18 +323,28 @@ Compiler.prototype = {
 
   Hash: function(hash) {
     let pairs = hash.pairs,
-        i = 0,
+        i = 0, splat = null,
```

Comment:
nit: This should be on a separate line, probably above `i` so `i` and `l` are declared next to one another.